src/ReaderScreen/components/AudioPlayer/hooks/useAudioManifest/index.js

```diff
@@ -34,44 +35,76 @@ const useAudioManifest = (baniID) => {
   };
 
   // Merge downloaded tracks with API tracks
-  const mergeDownloadedTracks = (apiTracks, downloadedTracks) => {
+  const mergeDownloadedTracks = async (apiTracks, downloadedTracks) => {
     if (!apiTracks || apiTracks.length === 0) {
       // If no API data, use downloaded tracks
       if (!downloadedTracks || downloadedTracks.length === 0) {
         return [];
       }
-      return downloadedTracks.map((track) => {
-        const fullLocalPath = `${DocumentDirectoryPath}/audio/${track.audioUrl}`;
-        const lyricsUrlPath = `${DocumentDirectoryPath}/audio/${track.lyricsUrl}`;
-        return {
-          id: track.id,
-          track_id: track.track_id,
-          artistID: track.artistID,
-          audioUrl: fullLocalPath,
-          displayName: track.displayName,
-          trackLengthSec: track.trackLengthSec,
-          trackSizeMB: track.trackSizeMB,
-          lyricsUrl: track.lyricsUrl ? lyricsUrlPath : null,
-        };
-      });
-    }
+      const validatedDownloads = await Promise.all(
```

Comment:
If even one of the `exists()` threw an error, the entire operation fails. Wrap each file existence check in try catch block to handle the error gracefully.